core/src/eval/value/lens.rs

```diff
@@ -283,6 +283,50 @@ impl TermContent {
     }
 }
 
+/// Generates a lens implementation with a constructor and an extractor for a given [Term]
+/// constructor and the corresponding type of its payload.
+///
+/// # Arguments
+///
+/// - `$lens_cons`: the name of the constructor
+/// - `$lens_extrct`: the  name of the extractor
+/// - `$term_cons`: the corresponding variant of [Term]
+/// - `$type`: the type of the argument of `$term_cons`
+///
+/// # Safety
+///
+/// The generated lens constructor is unsafe. The safety rule for the constructor is specified in
+/// the function documentation within the macro definition just below.
+macro_rules! impl_term_lens {
+    ( $lens_cons:ident, $lens_extrct:ident, $term_cons:ident, $type:ty ) => {
+        impl ValueLens<$type> {
+            // Creates a new lens extracting `Term::$term_cons`.
+            //
+            // # Safety
+            //
+            // `value` must be a value block with tag `DataTag::Term`, and the inner term must
+            // match `Term::$term_cons`.
```

Comment:
The inner term part isn't actually a safety issue, right? Because `unreachable!` isn't unsafe, it's just a panic...